glean-core/rlb/src/net/mod.rs

```diff
@@ -192,6 +204,9 @@ impl UploadManager {
         let thread = handle.take();
 
         if let Some(thread) = thread {
+            // poke the thread in case it's in `Wait`.
+            let _ = self.inner.tx.send(());
```

Comment:
fun fact, this would work just as well:

```suggestion
            _ = self.inner.tx.send(());
```

(though if clippy doesn't complain I don't complain)